design-documents/graph-ql/coverage/b2b/purchase-order.graphqls

```diff
@@ -0,0 +1,303 @@
+}
+
+interface PurchaseOrderApprovalRuleConditionInterface {
+    operator: PurchaseOrderApprovalRuleConditionOperator! @doc(description: "The operator to be used for evaluation of the approval rule condition")
+}
+
+enum PurchaseOrderApprovalRuleConditionOperator {
+    MORE_THAN
+    LESS_THAN
+    MORE_THAN_OR_EQUAL_TO
+    LESS_THAN_OR_EQUAL_TO
+}
+
+type PurchaseOrderApprovalRuleConditionAmount implements PurchaseOrderApprovalRuleConditionInterface {
+    amount: Money! @doc(description: "The amount to be be used for evaluation of the approval rule condition")
+}
+
+type PurchaseOrderApprovalRuleConditionQuantity implements PurchaseOrderApprovalRuleConditionInterface {
+    quantity: Int! @doc(description: "The quantity to be be used for evaluation of the approval rule condition")
+}
+
+enum PurchaseOrderApprovalRuleStatus {
+    ENABLED
+    DISABLED
+}
+
+enum PurchaseOrderApprovalRuleType {
+    ORDER_TOTAL
+    SHIPPING_COST
+    NUMBER_OF_SKUS
+}
+
+type PurchaseOrder {
+    uid: ID! @doc(description: "Unique identifier for the purcahse order")
+    number: String! @doc(description: "The purchase order number")
+    order: CustomerOrder @doc(description: "The reference to the order placed based on the purchase order")
+    purchase_order_date: String! @doc(description: "The date the purchase order was created")
+    created_by: String! @doc(description: "The name of the user who created the purchase order")
+    status: PurchaseOrderStatus! @doc(description: "The current status of the purcahse order")
+    total: PurchaseOrderTotal @doc(description: "Contains details about the calculated totals for the purchase order")
+    comments(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderComments @doc(description: "Purchase order comments")
+    payment_methods: [PaymentMethod] @doc(description: "Payment details for the purchase order")
+    shipping_address: CustomerAddress @doc(description: "The shipping address for the purchase order")
+    billing_address: CustomerAddress @doc(description: "The billing address for the purchase order")
+    carrier: String @doc(description: "The shipping carrier for the purchase order delivery")
+    shipping_method: String @doc(description: "The delivery method for the purchase order")
+    items(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderItems @doc(description: "Items that belong to the purchase order")
+    history_log(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderHistoryLog @doc(description: "The log of the events related to the purchase order")
+    approval_flow: PurchaseOrderApprovalFlow @doc(description: "The log of the events related to the purchase order approval flow")
+    available_actions: [PurchaseOrderAction] @doc(description: "Purcahse order actions available to the customer. Can be used to display action buttons on the client")
+}
+
+enum PurchaseOrderAction {
```

Comment:
There is also validate action, need to add corresponding mutation too.

Reply:
Should be added then, I have not seen it in the UI